cpg_workflows/stages/large_cohort.py

```diff
@@ -255,16 +272,242 @@ def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput:
         return self.make_outputs(cohort, outputs, job)
 
 
+@stage()
+class DenseBackground(CohortStage):
+    """
+    Will densify a background dataset such as 1KG, HGDP, or 1KG-HGDP
+    """
+
+    def expected_outputs(self, cohort: Cohort) -> dict[str, Path]:
+        background_datasets: list[str] = config_retrieve(['large_cohort', 'pca_background', 'datasets'], False)
+        background_storage_paths: dict[str, Path] = {
+            dataset: Path(config_retrieve(['storage', dataset, 'default'])) for dataset in background_datasets
+        }
+        return {
+            bg_dataset: background_storage_paths[bg_dataset] / 'dense' / 'dense.mt'
+            for bg_dataset in background_datasets
+        }
+
+    def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput:
+        # no datasets, no running
+        if not config_retrieve(['large_cohort', 'pca_background', 'datasets'], False):
+            return self.make_outputs(target=cohort)
+
+        # inputs
+        background_datasets: list[str] = config_retrieve(['large_cohort', 'pca_background', 'datasets'], False)
+        background_storage_paths: list[Path] = [
+            Path(config_retrieve(['storage', dataset, 'default']) for dataset in background_datasets),
+        ]
+
+        # construct input paths
+        # TODO: Allow for input background datasets to be '.mt' files
+        background_vds_version = dense_background_vds_version()
+        input_paths = {
+            bg_dataset: background_storage_paths[i] / 'vds' / f'{background_vds_version}.vds'
+            for i, bg_dataset in enumerate(background_datasets)
+        }
+
+        job = get_batch().new_job(f'Densify {", ".join(background_datasets)} background datasets')
+        job.storage('500Gi').image(config_retrieve(['workflow', 'driver_image']))
+
+        # localise the Dense MT
+        for bacgkround_vds in input_paths.values():
+            job.command(f'gcloud --no-user-output-enabled storage cp -r {bacgkround_vds} $BATCH_TMPDIR')
+
+        # expected outputs
+        outputs = self.expected_outputs(cohort)
+
+        # localise QC variants table
+        qc_variants_ht = config_retrieve(['references', 'ancestry', 'sites_table'])
+        job.command(f'gcloud --no-user-output-enabled storage cp -r {qc_variants_ht} $BATCH_TMPDIR')
```

Comment:
AFAIK you should be fine without localising the data for this stage - this is just reformatting/annotating the contents of a MT/VDS, and Query-on-batch should be able to handle that OK. 

The stages where we need Dataproc are where there's a feature that's not been ported to QOB yes, like PCA. To pull those out of Dataproc we need to run locally, but for this work we should be fine running in QOB